lib/msf/core/exploit/check_module.rb

```diff
@@ -0,0 +1,80 @@
+# -*- coding: binary -*-
+
+#
+# This mixin implements an exploit's check method by invoking an aux module
+#
+# NOTE: The module's run_host/run method MUST return an Msf::Exploit::CheckCode
+#
+
+module Msf
+module Exploit::Remote::CheckModule
+
+  def initialize(info = {})
+    super
+
+    register_advanced_options([
+      OptString.new('CheckModule', [true, 'Module to check with'])
+    ])
+  end
+
+  # When this mixin is included, this method becomes the exploit's check method
+  #
+  # @return [Msf::Exploit::CheckCode] Whether or not the target is vulnerable
+  def check
+    # Instantiate the module
+    mod = framework.modules.create(check_module)
+
+    # Bail if we couldn't
+    unless mod
+      print_error("Could not instantiate #{check_module}")
+      return CheckCode::Unsupported
+    end
+
+    # Bail if run_host/run isn't defined
+    if mod.respond_to?(:run_host)
+      meth = :run_host
```

Comment:
`Msf::Auxiliary::Scanner#run` now returns results, so a separate check for `run_host` isn't needed.

Reply:
Good point. `run` is the entry point for `run_host`, anyway. This check is unnecessary.

Reply:
(As of #12509.)